packages/server/package.json

```diff
@@ -105,8 +106,10 @@
     "syntax-error": "1.4.0",
     "systeminformation": "4.21.1",
     "term-size": "2.1.0",
+    "through2": "^2.0.0",
     "tough-cookie": "3.0.1",
     "trash": "5.2.0",
+    "ts-node": "8.5.4",
```

Comment:
What kind of size does this add to Cypress's binary?

Reply:
Also, what happens if a user is using a different version of TypeScript that isn't compatible with the one we ship? Can they configure it to use their version instead ours for their plugins file?

Reply:
1\. [According to bundlephobia](https://bundlephobia.com/result?p=ts-node@8.8.1), it adds about 50kb. 

2\. 

This PR uses the typescript in the user's project, not secretly bundled one in Cypress that'll be eventually removed in #6051. Check the code below.

https://github.com/cypress-io/cypress/blob/29cff450dd23cd68dc01aaa1ca7edc52ed5f0ec5/packages/server/lib/project.js#L105-L118

[And according to the code of `ts-node`](https://github.com/TypeStrong/ts-node/blob/master/src/index.ts#L137-L142), it uses the typescript compiler api. `ts-node` here cannot compile ts files only if typescript team decides to break the typescript compiler api. It's very unlikely.